tests/FSharp.Compiler.ComponentTests/Conformance/UnionTypes/MultiCaseUnionStructTypes.fs

```diff
@@ -0,0 +1,250 @@
+﻿namespace FSharp.Compiler.ComponentTests.Conformance
+
+open Xunit
+open FSharp.Test.Compiler
+
+module MultiCaseUnionStructTypes =
```

Comment:
Please add a test for field names assigning, but having conflicts.
1. One testconflicting names within one case
2. second test conflicting names across cases
3. third test a conflict between user-provided name called "item" and the auto-generated one called "item" => that one actually works now, because compiler puts in an underscore: https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AbEAzAzmgExAGoAfAOwEMBbGXAB0rBgAIAxCCAWACgBtADwBlAC5QArmBEA+ALq8RAT3qtREqQFVyASwjkWAXl4sTLUiwCCLCNhPbyI46fMAha7e0iY1EPZFA=

Reply:
I have added unit tests for this I believe :)